src/gei/Commands/MigrateSecretScanningAlertsCommand.cs

```diff
@@ -0,0 +1,196 @@
+        var githubTargetOrg = new Option<string>("--github-target-org")
+        {
+            IsRequired = true, Description = "Uses GH_PAT env variable or --github-target-pat option."
+        };
+        var targetRepo = new Option<string>("--target-repo")
+        {
+            IsRequired = false, Description = "Defaults to the name of source-repo"
+        };
+        var targetApiUrl = new Option<string>("--target-api-url")
+        {
+            IsRequired = false,
+            Description =
+                "The URL of the target API, if not migrating to github.com. Defaults to https://api.github.com"
+        };
+
+        // GHES migration path
+        var ghesApiUrl = new Option<string>("--ghes-api-url")
+        {
+            IsRequired = false,
+            Description =
+                "Required if migrating from GHES. The API endpoint for your GHES instance. For example: http(s)://ghes.contoso.com/api/v3"
+        };
+        var noSslVerify = new Option("--no-ssl-verify")
+        {
+            IsRequired = false,
+            Description =
+                "Only effective if migrating from GHES. Disables SSL verification when communicating with your GHES instance. All other migration steps will continue to verify SSL. If your GHES instance has a self-signed SSL certificate then setting this flag will allow data to be extracted."
+        };
+        var githubSourcePat = new Option<string>("--github-source-pat") { IsRequired = false };
+        var githubTargetPat = new Option<string>("--github-target-pat") { IsRequired = false };
+        var verbose = new Option("--verbose") { IsRequired = false };
+        var dryRun = new Option("--dry-run")
+        {
+            IsRequired = false,
+            Description =
+                "Execute in dry run mode to see what secrets will be matched and changes applied, but do not make any actual changes."
+        };
+
+        AddOption(githubSourceOrg);
+        AddOption(sourceRepo);
+        AddOption(githubTargetOrg);
+        AddOption(targetRepo);
+        AddOption(targetApiUrl);
+
+        AddOption(ghesApiUrl);
+        AddOption(noSslVerify);
+
+        AddOption(githubSourcePat);
+        AddOption(githubTargetPat);
+        AddOption(verbose);
+        AddOption(dryRun);
+
+        Handler = CommandHandler.Create<MigrateSecretScanningAlertsCommandArgs>(Invoke);
+    }
+
+    public async Task Invoke(MigrateSecretScanningAlertsCommandArgs args)
+    {
+        if (args is null)
+        {
+            throw new ArgumentNullException(nameof(args));
+        }
+
+        _log.Verbose = args.Verbose;
+
+        LogAndValidateOptions(args);
+
+        var sourceGitHubApi = _sourceGithubApiFactory.Create(args.GhesApiUrl, GetSourceToken(args));
+        var targetGithubApi = _targetGithubApiFactory.Create(args.TargetApiUrl, GetTargetToken(args));
+        var migrationService = _secretScanningAlertServiceFactory.Create(sourceGitHubApi, targetGithubApi);
```

Comment:
Three is no need to inject source and target Github APIs just to create the `secretScanningAlertService`, the factory can easily create everything that is needed so you can use it like this:
```c#
var migrationService = _secretScanningAlertServiceFactory.Create(args.GhesApiUrl, GetSourceToken(args), args.TargetApiUrl, GetTargetToken(args));
```